torch/csrc/jit/backends/backend_debug_handler.cpp

```diff
@@ -3,8 +3,12 @@
 namespace torch {
 namespace jit {
 
+namespace {
+thread_local BackendDebugHandleManager* debug_handle_manager_ptr{nullptr};
```

Comment:
Yes, that was the reason. I dont anticipate that this will be a general case but there is nothing in the API that restricts client to load libtorch, launch threads and do lowering via different threads. So I did not want to make assumptions I could not assert against.